src/videojs.ads.js

```diff
@@ -43,13 +43,35 @@ var
   },
 
   /**
+   * Returns a boolean indicating if given player is in live mode.
+   */
+  isLive = function(player) {
+    if (player.duration() === Infinity) {
+      return true;
+    } else if (videojs.browser.IOS_VERSION === "8" && player.duration() === 0) {
```

Comment:
Sounds like iOS is once again doing stuff that is outside of the spec (https://html.spec.whatwg.org/multipage/embedded-content.html#dom-media-duration) however, yes, that's totally something that videojs itself should be handling.